### PR TITLE
v2 add role and cluster role binding

### PR DESCRIPTION
**Pull request by zreigz**
**What this PR does / why we need it**: v2 add role and cluster role binding

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

```release-note
v2 adds role and cluster role binding:
POST /api/v2/projects/{project_id}/clusters/{cluster_id}/clusterroles/{role_id}/clusterbindings
POST /api/v2/projects/{project_id}/clusters/{cluster_id}/roles/{namespace}/{role_id}/bindings

```

### CODE REVIEW

**Comment by zreigz** [comment]
@lsviben PTAL

**Review by lsviben — COMMENTED** [review]
/lgtm
/approve

**Comment by zreigz** [comment]
@moadqassem PTAL

File: pkg/handler/common/binding.go [hunk 1]
```diff
@@ -0,0 +1,220 @@
...
+	apiv1 "k8c.io/kubermatic/v2/pkg/api/v1"
+	"k8c.io/kubermatic/v2/pkg/handler/middleware"
+	"k8c.io/kubermatic/v2/pkg/handler/v1/common"
+	"k8c.io/kubermatic/v2/pkg/provider"
+	"k8c.io/kubermatic/v2/pkg/util/errors"
+
+	rbacv1 "k8s.io/api/rbac/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/rand"
+
```
Comment (moadqassem) [inline pkg/handler/common/binding.go]: Nit: `sig.k8s` is a kubernetes package so you can merge it with the other packages.

File: pkg/handler/common/binding.go [hunk 2]
```diff
@@ -0,0 +1,220 @@
...
+	}
+
+	if existingClusterRoleBinding == nil {
+		return nil, fmt.Errorf("the cluster role binding not found")
+	}
+
+	oldBinding := existingClusterRoleBinding.DeepCopy()
+	for _, subject := range existingClusterRoleBinding.Subjects {
+		if clusterRoleUser.UserEmail != "" && subject.Name == clusterRoleUser.UserEmail {
+			return nil, errors.NewBadRequest("user %s already connected to role %s", clusterRoleUser.UserEmail, roleID)
```
Comment (moadqassem) [inline pkg/handler/common/binding.go]: Shouldn't this be `return nil, errors.NewBadRequest("user %s already connected to cluster role %s", clusterRoleUser.UserEmail, roleID)`

File: pkg/handler/common/binding.go [hunk 3]
```diff
@@ -0,0 +1,220 @@
...
+		}
+		if clusterRoleUser.Group != "" && subject.Name == clusterRoleUser.Group {
+			return nil, errors.NewBadRequest("group %s already connected to role %s", clusterRoleUser.Group, roleID)
```
Comment (moadqassem) [inline pkg/handler/common/binding.go]: return nil, errors.NewBadRequest("group %s already connected to cluster role %s", clusterRoleUser.Group, roleID)

File: pkg/handler/common/binding.go (hunk 1 above)
Comment (zreigz) [reply pkg/handler/common/binding.go]: done

File: pkg/handler/common/binding.go (hunk 2 above)
Comment (zreigz) [reply pkg/handler/common/binding.go]: done

File: pkg/handler/common/binding.go (hunk 3 above)
Comment (zreigz) [reply pkg/handler/common/binding.go]: done

**Comment by zreigz** [comment]
@moadqassem PTAL

**Pull request merged**